### DISCUSSION TITLE
[A-L] (2023/24) Foglio 6 - Esercizio 4

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/algebra#314] [Esercizi - Viaggi/Piazza]
![Screenshot 2023-11-09 172728](https://github.com/sapienzastudentsnetwork/algebra2324/assets/107944476/ce2e4c5b-b9e5-4368-a2ce-467ff68eb9c2)

**Comment by daromacarlo** [comment]
![Immagine 2023-12-14 171042](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/8e8f9a00-0276-44d1-a57a-0cf146576607)
![Immagine 2023-12-14 171100](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/13311d04-fd7b-4e02-a80a-ae63a50543ac)
![Immagine 2023-12-14 171116](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/46eff3e0-2e6d-4135-aa6d-aa8304b2fc0b)

**Comment by daromacarlo** [comment]
teorema usato
![Immagine 2023-12-14 171502](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/2a022c58-0b4d-496d-97af-a9eeb1c7d906)

ho un grande dubbio, quando un esercizio chiede dimostra che... sta implicitamente dicendo che la sua affermazione è vera e quindi bisogna dimostrare che è vera oppure come nel mio caso posso affermare che è falso?
grazie

**Comment by daromacarlo** [comment]
punto 1)
![Immagine 2023-12-14 184427](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/2ae2c412-da58-4a8f-9ed3-b1cece953dbd)
![Immagine 2023-12-14 184443](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/08e052b2-eb2a-42dd-a3a8-7698470593a8)
![Immagine 2023-12-14 184455](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/69bce1c1-9590-4e00-8668-8d0596b0cfdb)

**Comment by daromacarlo** [comment]
punto 2)
![Immagine 2023-12-14 185827](https://github.com/sapienzastudentsnetwork/algebra2324/assets/135441131/6b0b501e-e2ab-488f-aba7-2dedfedd3559)

**Reply by interestingmaneuver** [reply]
Manca il sottogruppo di ordine 5, cioè $\langle 2^{20/5} \rangle = \langle 2^4 \rangle$, poi questo elenco è esaustivo solo perché $U_{25}$ è ciclico.

**Reply by daromacarlo** [reply]
hai ragione, grazie

**Comment by CiottoloMaggico** [comment]
![image](https://github.com/sapienzastudentsnetwork/algebra2324/assets/79638284/79062666-35a4-4886-97c9-75555be97d3a)

**Comment by Elia-Belli** [comment]
**(1)** Per dimostrare che $(\mathcal{U}(\mathbb{Z}\_{25}),\cdot)$ è ciclico cerchiamo un elemento $g$ che lo generi, ovvero tale che $\langle g\rangle=\mathcal{U}(\mathbb{Z}_{25})$.

Osserviamo innanzitutto che $|\mathcal{U}(\mathbb{Z}\_{25})|=\phi(25)=20$ quindi anche l'ordine di $|\langle g\rangle|$ deve essere $20$
Quindi mettiamoci alla ricerca di un sottogruppo generato da $g\in \mathcal{U}(\mathbb{Z}\_{25})$ di ordine $20$

> Ricordiamo che l'ordine di un elemento è il minimo numero tale che $g^{o(g)}=1$, quindi non basta che $g^{20}=1$ (che in questo caso è vero per tutti gli elementi) ma $20$ deve essere il minimo numero per cui ciò avviene, il Teorema di Lagrange ci dice chi sono i suoi possibili competitors.

Il Teorema di Lagrange fissa i possibili ordini dei sottogruppi di $\mathcal{U}(\mathbb{Z}\_{25})$ ai divisori di $20$ ovvero $\\{1,2,4,5,10,20\\}$
Ora tocca testare gli elementi finchè non ne troviamo uno di ordine $20$ (se esiste), $\langle 1\rangle$ è il sottogruppo banale $\\{1\\}$ partiamo quindi da $2$:

$$2^1\neq 1;\ 2^2\neq 1;\ 2^4\neq 1;\ 2^5\neq 1;\ 2^{10}\neq 1; 2^{20}=1\ in\ \mathbb{Z}\_{25}$$

Che fortuna! Abbiamo trovato un generatore al primo colpo $\langle 2\rangle=\mathcal{U}(\mathbb{Z}_{25})$ allora è effettivamente ciclico.

> Non è detto che il generatore sia unico. Inoltre mi domando se ci sia un modo per dire a priori se il gruppo è ciclico senza dover testare, nel caso peggiore, tutti gli elementi del gruppo (meno il neutro), il che ha una complessità di $O(\phi(n))$

**(2)** Per determinare i sottogruppi ricordiamo il Teorema di Struttura per gruppi ciclici che stabilisce corrispondenza biunivoca tra i divisori di $20$ e i sottogruppi di $<2>$, in particolare i sottogruppi sono tutti e soli $<2^{\frac{20}{k}}>$ con $k$ divisore di $20$:

$$<2>\ <4>\ <16>\  <7>\ <24>\ <1>$$

**Reply by interestingmaneuver** [reply]
> Inoltre mi domando se ci sia un modo per dire a priori se il gruppo è ciclico senza dover testare

Per alcuni $U_n$ e $\mathbb{Z}_n$ si può usare il teorema cinese dei resti:

Dati $n_1, \ldots, n_r \in \mathbb{Z}$ a due a due coprimi, $f: \mathbb{Z}\_{n\_1 \ldots n\_r} \to \mathbb{Z}\_{n\_1} \times \ldots \times \mathbb{Z}\_{n\_r}$ data da $x \mod n\_1 \ldots n\_r \mapsto (x \mod n\_1, \ldots, x \mod n\_r)$ è un omomorfismo di gruppi banalmente invertibile, quindi un isomorfismo.

Anche sul gruppo delle unità è un isomorfismo, dato $x \in \mathbb{Z}_n$, $x \in U_n \iff (x, n) = 1$. Allora se $(x, n) = 1$, anche $(x, n_i) = 1$ per $1 \le i \le r$.

Ad esempio:

- $U_{12} \cong U\_{3} \times U\_{4}$: $U\_3$ è ciclico di ordine 2, quindi $U\_3 \cong \mathbb{Z}\_2$, $U\_4$ è anch'esso ciclico di ordine 2, quindi $U\_4 \cong \mathbb{Z}\_2$. Però $\mathbb{Z}\_2 \times \mathbb{Z}\_2$ non è ciclico, quindi $U_{12}$ non è ciclico.
- $U_{14} \cong U_{2} \times U_{7}$. $U_2$ è banalmente ciclico di ordine 1, $U_7$ è ciclico di ordine 6. $U_2 \times U_7$ è ciclico essendo $(1, 6) = 1$, quindi $U_{14}$ è ciclico.

Attenzione, non è detto che $U_{25} \cong U_{5} \times U_{5}$, perché $(5, 5) = 5 \ne 1$, quindi il teorema cinese dei resti non si può applicare in questo esercizio.

**Reply by interestingmaneuver** [reply]
In aggiunta, sapendo ad esempio che $U_{14} \cong U_2 \times U_7$, e che:

* Un generatore di $U_2$ è $[1]$
* Un generatore di $U_7$ è $[3]$

(quindi $([1], [3])$ è un generatore di $U_2 \times U_7$)

Dall'isomorfismo descritto sopra si ha che un generatore di $U_{14}$ è la soluzione di

$$
\begin{cases}
x \equiv 1 \mod 2\\
x \equiv 3 \mod 7
\end{cases}
$$

ossia $3 \mod 14$, quindi $[3]$.
